docs/authelia-from-scratch.md

```diff
@@ -0,0 +1,461 @@
+# Installing **Authelia** on Debian bare metal from scratch
+
+This document describes how to build **Authelia** from scratch and how to
+create a basic working configuration.
+
+For the purpose of this document the following examples are used;
+
+    This is the hostname people will access when they want to access the protected site;
+    landing.example.com
+    Hostname of the Authelia service;
+    authelia.example.com
+    Hostname of the protected site;
+    protected.example.com
+    $editor to be replaced by your favourite editor
+    $user to be replaced by the username you want to create to login to landing.example.com
+
+The following passwords are needed;
+
+    slapd-admin-password=<slapd-admin-password>
+    redis-password=<redis-password>
+    mariadb-password=<mariadb-password>
+    user-password=<user-password>
+    jwtsecret=<jwt-secret>
+    sessionsecret=<session-secret>
+    
+These packages are optional and only needed if you are planning to compile from source;
+
+    * GO
+    * Docker
+    * Docker Compose
+    * Node.JS
+    and you can then skip the section 'Download, Build and Install Authelia'
+
+**NOTE** [you can download the binary from https://github.com/clems4ever/authelia/releases/](https://github.com/clems4ever/authelia/releases/)
+
+This document assumes you have a domain-wildcard SSL certificate in /etc/nginx/ssl/
+
+    # cp -aL /etc/letsencrypt/live/example.com/fullchain.pem /etc/nginx/ssl/xxx.example.com.fullchain
+    # cp -aL /etc/letsencrypt/live/example.com/privkey.pem /etc/nginx/ssl/xxx.example.com.key
+    # cp -aL /etc/letsencrypt/live/example.com/cert.pem /etc/nginx/ssl/xxx.example.com.cert
+
+**NOTE** First copy the /etc/letsencrypt/live/example.com directory from the location where you have letsencrypt running
+
+**NOTE** All installation steps below are to be executed as root or as a normal user using 'sudo'
+
+### Install Debian the latest version of Debian 10
+    Debian Network install from a minimal CD: https://www.debian.org/CD/netinst/
+    
+### Uninstall apparmor and install your favourite editor if not already installed
+    # apt-get remove apparmor
+    # apt-get install $editor gcc g++ make slapd ldap-utils redis-server curl mariadb-server nginx git apt-transport-https ca-certificates gnupg2 software-properties-common ntpdate
+
+**NOTE** You will be prompted to set the $slapd-admin-password
+
+#### Install GO;
```

Comment:
Do you really need that at this point if the user does not build from source?

Reply:
Maybe we can mark it as optional in that case. What do you think?

Reply:
It's already on the optional list :)